ui-lib/src/main/java/cash/z/ecc/ui/screen/backup/model/BackupStage.kt

```diff
@@ -0,0 +1,39 @@
+package cash.z.ecc.ui.screen.backup.model
+
+import cash.z.ecc.ui.screen.onboarding.model.Index
+import cash.z.ecc.ui.screen.onboarding.model.Progress
+
+enum class BackupStage {
```

Comment:
I understand this is kind of a state machine for the initial flow, isn't it?

We are handling these two things separately. Onboarding is its own flow and Recovery Phrase Backup/Display/ Backup validation is another one.

Mostly because:
The onboarding can change over time and be shown to existing users that already have a wallet set, backed up and validated.
The User can request to see the recovery phrase later in time.

Reply:
As of this PR, there are effectively 3 state machines right now in the codebase:
 - Wallet (none, needs backup, ready)
 - Onboarding (used when when wallet state is none)
 - Backup (used when wallet state is needs backup)

Onboarding and Backup could be re-displayed in the future if desired.  I separated them, so they could be updated and modified independently without as much complex interdependency.  I think we're saying the same thing.

The line of code indicated by your comment is a state machine for the backup flow only.  

Realistically, I expect we'll have a separate screen for the viewing of the recovery phrase in the future in a non-backup use case.  It will probably re-use some of the Composables, but it probably won't re-use the state machine.

Does that answer your question?

Reply:
Yes!

Reply:
Wallet: https://github.com/zcash/secant-android-wallet/blob/f47bf2fcf104f2b318d2b218558dff7314676684/ui-lib/src/main/java/cash/z/ecc/ui/screen/home/viewmodel/WalletViewModel.kt#L116
Onboarding: https://github.com/zcash/secant-android-wallet/blob/f47bf2fcf104f2b318d2b218558dff7314676684/ui-lib/src/main/java/cash/z/ecc/ui/screen/onboarding/model/OnboardingStage.kt

What makes it possibly slightly confusing is the need for 3 files for the Onboarding and Backup state machines.  Each follows a pattern of the enum of steps <https://github.com/zcash/secant-android-wallet/blob/f47bf2fcf104f2b318d2b218558dff7314676684/ui-lib/src/main/java/cash/z/ecc/ui/screen/onboarding/model/OnboardingStage.kt>, there's the mutable observable object <https://github.com/zcash/secant-android-wallet/blob/f47bf2fcf104f2b318d2b218558dff7314676684/ui-lib/src/main/java/cash/z/ecc/ui/screen/onboarding/state/OnboardingState.kt>, then there's the viewmodel for Android lifecycle state retention <https://github.com/zcash/secant-android-wallet/blob/f47bf2fcf104f2b318d2b218558dff7314676684/ui-lib/src/main/java/cash/z/ecc/ui/screen/onboarding/viewmodel/OnboardingViewModel.kt>